lib/index.ts

```diff
@@ -0,0 +1,572 @@
+/*
+Copyright 2017 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as request from 'request-promise';
+
+const FrontURL = 'https://api2.frontapp.com';
+
+interface FrontRequest {
+	reqType: string;
+	path: string;
+}
+
+type InternalCallback = (err: Error | null, response: any | null) => void;
+
+export class Front {
+	public comment: {
+		create: (params: FrontCommentRequest.Create, callback?: FrontCallback<FrontComment>) => Promise<FrontComment>;
```

Comment:
Not sure if it's possible, but it'd be a lot better IMO, both for code readability and client usage of the SDK, if types were namespaced such that the `Front` prefix could be removed from the definitions -- i.e. `FrontCommentRequest` -> `Front.CommentRequest` (or `Front.Comment.Request`) -> `CommentRequest`.